### PR TITLE
Update dream.py. k_euler_a and k_dpm_2_a M1 fix

### PR DESCRIPTION
**Pull request by Any-Winter-4079**
Make results reproducible (so runs with the same seed produce the same result). Implements fix by @wbowling referenced in https://github.com/lstein/stable-diffusion/issues/397#issuecomment-1240679294
Example: `"banana sushi" -s50 -W512 -H512 -C7.5 -Ak_euler_a -S618376249`
Before
Screenshot 2022-09-14 at 23 27 27
After
Screenshot 2022-09-14 at 23 27 45

Fix should be ignored in non-MPS devices, but test to make sure.

### CODE REVIEW

File: scripts/dream.py [hunk 1]
```diff
@@ -14,12 +14,30 @@
 from ldm.dream.server import DreamServer, ThreadingDreamServer
 from ldm.dream.image_util import make_grid
 from omegaconf import OmegaConf
+import torch
```
Comment (lstein) [inline scripts/dream.py:17]: The torch library takes a significant time to load; it's particularly a problem in my environment where the libraries are being loaded from a shared network mount. I don't like having to load torch just to print out the `dream.py -h` help message, and for this reason torch and other big libraries are only imported later on in generate.py.

File: scripts/dream.py [hunk 2]
```diff
@@ -14,12 +14,30 @@
...
 
 # Placeholder to be replaced with proper class that tracks the
 # outputs and associates with the prompt that generated them.
 # Just want to get the formatting look right for now.
 output_cntr = 0
 
+def fix_func(orig):
```
Comment (lstein) [inline scripts/dream.py:24]: It feels like this chunk of code should be moved into ldm.generate. Otherwise people who are writing scripts against Generate won't get the benefits of your fix.

**Review by lstein — CHANGES_REQUESTED** [review]
Please see two comments.

**Review by Any-Winter-4079 — COMMENTED** [review]
I've removed it from `dream.py` and added it in `generate.py`

**Review by lstein — APPROVED** [review]

**Comment by lstein** [comment]
Thank you!
